src/mpyl/utilities/parallel/__init__.py

```diff
@@ -0,0 +1,35 @@
+"""Utility tool for running commands in parallel"""
+from concurrent.futures import Future, ThreadPoolExecutor
+from dataclasses import dataclass
+from typing import Callable, Any, TypeVar, Iterable
+
+T = TypeVar("T")
+
+
+@dataclass(frozen=True)
+class ParallelObject:
+    function: Callable
+    parameters: dict[Any, Any]
```

Comment:
I guess the keys in this dict are always strings?

Reply:
Yes indeed, there was a point where they weren't and I forgot the change it back.